web/src/hooks/useAllChallenges.ts

```diff
@@ -0,0 +1,84 @@
+import { readContract } from '@wagmi/core';
+import * as trackerContract from '@/contracts/tracker';
+import { useState, useEffect } from 'react';
+import { wagmiConfig as config } from '@/OnchainProviders';
+import { usePublicClient } from 'wagmi';
+import { Address } from 'viem';
+import { Challenge, ChallengeDetail } from '@/types';
+import { challengeMetaDatas } from '@/constants';
+
+const useAllChallenges = () => {
+  const publicClient = usePublicClient({ config });
+
+  const [loading, setLoading] = useState(true);
+  const [challenges, setChallenges] = useState<Challenge[] | []>([]);
+  const [error, setError] = useState<unknown | null>(null);
+
+  useEffect(() => {
+    if (!publicClient?.multicall) return;
+
+    const fetchData = async () => {
+      try {
+        setLoading(true);
+
+        const challengeCount = await readContract(config, {
+          abi: trackerContract.abi,
+          address: trackerContract.address,
+          functionName: 'challengeCounter',
+        });
+
+        const result = await publicClient.multicall({
+          contracts: Array.from({ length: Number(challengeCount.toString()) }, (_, i) => ({
+            address: trackerContract.address,
+            abi: trackerContract.abi,
+            functionName: 'challenges',
+            args: [i],
+          })),
+        });
+
+        const newData: Challenge[] = result
+          .map((raw, idx) => {
+            const res = raw.result as any as [
+              Address,
+              bigint,
+              bigint,
+              bigint,
+              Address,
+              bigint,
+              bigint,
+              boolean,
+            ];
+            return {
+              id: BigInt(idx),
+              verifier: res[0],
+              targetNum: Number(res[1].toString()),
+              startTimestamp: Number(res[2].toString()),
+              endTimestamp: Number(res[3].toString()),
+              // donationOrg?: res[4];
```

Comment:
Do we want to remove this? Or save this for future use?

Reply:
kept donationDestination as address for now, we can add our own whitelist later!